packages/core/src/createMachine.ts

```diff
@@ -91,6 +128,10 @@ export function createMachine<
     TOutput,
     TTypesMeta
   >,
+  /**
+   * @internal
+   * @deprecated Use `setup({ ... })` or `machine.provide({ ... })` instead
+   */
```

Comment:
You can't add JSDocs directly to parameters. They would have to be hoisted to the JSDoc for the function declaration (and annotated through `@params` and such). Adding `@internal` could be considered a breaking change. But anyway, I'm not really sure if either of those is supported for params - since you have to hoist those for them to actually have an effect... they start to be applied to the function declaration itself and not to individual parameters.